homeassistant/components/stream/__init__.py

```diff
@@ -388,3 +392,7 @@ async def async_record(
             # Wait for latest segment, then add the lookback
             await hls.recv()
             recorder.prepend(list(hls.get_segments())[-num_segments:])
+
+    async def get_image(self) -> bytes:
+        """Fetch an image from the Stream and return it as a jpeg in bytes."""
+        return await self._keyframe_converter.get_image()
```

Comment:
I feel like this deserves a health check before running, e.g. is the worker thread started? maybe also pre-checking "self.available"? Or do you consider this the callers responsibility?

(either way i assume we'll want something faster than a 60 second timeout, which seems more like a failsafe in case things get stuck)

Reply:
I initially thought the caller should be responsible for this. It seems wasteful to have to check it every time, and the caller should generally know the status of stream.
However, it might be easier to actually take care of all this here in `stream` than to have platforms try and manage things. The problem then becomes managing the stream when given these image calls - I already saw a previous request to be able to lower the default stream timeout. I'm worried that introducing this separate use case  for stream will have a separate list of demands including a customizable stream timeout.
Yes the 60 second timeout was based on waiting for the next keyframe, which could be some time in the future with certain codecs/settings. Not sure exactly what we should drop it to though. If this was purely image generation, something like 0.5 seconds would probably work. However, this is the time from the request until the worker replies, which requires the worker to have gotten a new packet from its input. For RTSP feeds this extra time waiting for a packet is probably negligible, but if it's some cloud feed like a hls feed this might be like 5-10 seconds. I guess this is one argument for moving the image generation process back out of the worker.